app/views/shared/_char_count.html.erb

```diff
@@ -0,0 +1,28 @@
+<%# 
+  Reusable helper view for character count requirements.
+
+  Variables:
+    cur : current number of characters (default 0)
+    max : maximum number of characters allowed (default 255)
+    min : minimum number of characters allowed (default 0)
+    threshold : fraction of max to show the count at (default 0.75)
+    type : character count type (e.g.: post-title)
+%>
+
+<%
+  # defaults & normalization
+  cur ||= defined?(cur) && !cur.nil? ? cur.to_i : 0
+  max ||= defined?(max) && !max.nil? ? max.to_i : 255
+  min ||= defined?(min) && !min.nil? ? min.to_i : 0
+  threshold ||= defined?(threshold) && !threshold.nil? ? threshold.to_f : 0.75
+%>
+
+<span class="has-float-right has-font-size-caption js-character-count-<%= type %>"
```

Comment:
question: Am I missing something, or is the `type` not actually used anywhere? Do we need it?

also: the JS triggers on elements with a `data-character-count` attribute, which this doesn't have - what am I missing here?

Reply:
As for the `data-character-count` attribute - we actually do have it :) It's just set with an underscore, IIRC: https://github.com/codidact/qpixel/blob/47a4d2153851bb5009774f0535e0cae3012a8c39/app/views/posts/_form.html.erb#L65

Reply:
Ah, okay, I misunderstood the view. So this partial is _only_ the actual counter part - you still have to set the data attribute on the relevant form field separately?

Reply:
@ArtOfCode- yup, that's how it currently works as the event listener is delegated. That said, I am open to extending the scope of the PR a little further and devise a way to get rid of the necessity to set `data-character-count`  the class name matching the one of the count (also related to removing the need for the `type` parameter).

Reply:
@ArtOfCode-  are you still requesting a change here or is this thread resolved?